tools/firmware-utils/src/mktplinkfw.c

```diff
@@ -132,6 +132,12 @@ static struct flash_layout layouts[] = {
 		.kernel_la	= 0x80060000,
 		.kernel_ep	= 0x80060000,
 		.rootfs_ofs	= 0x100000,
+	}, {
+		.id		= "4Mmtk",
+		.fw_max_len	= 0x3a0000,
+		.kernel_la	= 0x80000000,
+		.kernel_ep	= 0x80000000,
```

Comment:
For 8Mmtk, this is different from kernel_la ...

Reply:
Actually, the 8Mmtk devices (AFAIK it's only the RE200v1) might boot with the default entry point as well. When adding support, I used the entry point TP-Link is using, but I don't think I tested the default entry point (I remember having problems with LZMA compression so I played around quite a bit).
Unfortunately, I glued my device back together and I don't want to risk a bricked device - no recovery mode present in U-Boot...

Reply:
Thanks for the shared info. U-Boot won't be much different on WA850RE v4 ...

Reply:
The entry point of 0x80060000 as was in the 4Mlzma format didnt work, and then i looked at how the TFTP to RAM method booted and noticed it using this address and tried it and it worked.
If i have the energy maybe i can take more of a look into how the vendor fw boots...

Reply:
Well, if it works reproducibly then I'm fine. I just wanted to bring this on the table for discussion, in case somebody else has could share some insights.